src/targets/gpu/device/where.cpp

```diff
@@ -0,0 +1,22 @@
+#include <migraphx/gpu/device/where.hpp>
+#include <migraphx/gpu/device/nary.hpp>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+namespace gpu {
+namespace device {
+
+void where(hipStream_t stream,
+           const argument& result,
+           const argument& arg0,
+           const argument& arg1,
+           const argument& arg2)
+{
+    nary(stream, result, arg0, arg1, arg2)([](auto condition, auto x, auto y)
```

Comment:
nary requires all args to be the same type, so all gpu related unit tests failed.

Reply:
So should this be implemented without using nary? Or could we maybe convert arg0 to the type of x in the parser and have the lambda as `{ return float_equal(condition, 1.0f) ? x : y; }` ?

Reply:
Ah yea it looks like `nary` cant be used here.